docs/modules/ROOT/pages/adr/0033-event-based-billing-oddo.adoc

```diff
@@ -0,0 +1,233 @@
+= ADR 0033 - Event-Based Billing
+:adr_author:    Gabriel Saratura
+:adr_owner:     Schedar
+:adr_reviewers:
+:adr_date:      2025-06-02
+:adr_upd_date:  2025-06-02
+:adr_status:    draft
+:adr_tags:      billing,odoo
+
+include::partial$adr-meta.adoc[]
+
+[NOTE]
+.Summary
+====
+For sending billing events to Odoo (https://docs.central.vshn.ch/event-billing.html[Event Based Billing^]), will use the existing AppCat controller to manage billing events and Kubernetes CRDs as the persistence mechanism.
+====
+
+== Requirements
+
+. Reuse existing tools and/or components of AppCat to build a resilient event-based billing solution.
+. Enable resending data to Odoo (historical record retention is required).
+. Persist the state of each billing request.
+. Do not block service creation, deletion, or update due to errors in the billing system.
+. [[odoo-sync-state]] Provide a way to verify sync state between Odoo and the Kubernetes cluster.
+. Do not lose any event (created, deleted, scaled), as this has a direct financial impact.
+
+
+== Solution Options
+
+After careful evaluation, the two most promising solutions for implementing event-based billing are:
+
+. <<AppCat Controller>>
+. <<Runtime Library via Crossplane Composition Functions>>
+
+=== AppCat Controller
+
+Transitioning from metered to event-based billing requires leveraging Kubernetes controllers more extensively.
+Our existing AppCat controller already handles event forwarding and webhooks, making it a natural candidate for integrating billing logic.
+
+*Pros:*
+
+* Full control over the lifecycle of VSHN custom resource services.
+* Customizable retry logic.
+* Clear separation of concerns between billing and service management.
+* Flexible support for different persistence backends.
+
+*Cons:*
+
+* More difficult to block or revert service operations when billing events fail to post to Odoo.
```

Comment:
Why is that a con? In requirements we write that we don't want that anyway.

Reply:
Indeed you are right, I will remove this. I followed general pros and cons without considering the requirements.